src/core/main.c

```diff
@@ -191,6 +192,30 @@ static int manager_find_user_config_paths(char ***ret_files, char ***ret_dirs) {
         return 0;
 }
 
+static int console_enable_line_wrapping(int tty_fd) {
+        int r;
+
+        assert(tty_fd >= 0);
+
+        if (terminal_is_dumb())
+                return 0;
+
+        r = fd_nonblock(tty_fd, true);
+        if (r < 0)
+                return log_debug_errno(r, "Failed to set /dev/console to non-blocking mode: %m");
+
+        /* Enable line wrapping. */
+        (void) loop_write_full(tty_fd, "\033[?7h", SIZE_MAX, 50 * USEC_PER_MSEC);
```

Comment:
The error is already ignored on caller's side, so maybe check for error here?

Reply:
Then I have to add a second error variable because we need the value from the first one to figure out whether we need to set back to blocking mode, so I'd prefer to keep it like this.

Reply:
But it feels really strange that errors returned are only from auxiliary calls. Adding a variable is not too much work? I.e.:

```c
int r, k;

...

k = fd_nonblock(fd, true);
if (k < 0)
        return log_debug_errno(...);

r = loop_write_full(...);

if (k > 0) {
        k = fd_nonblock(fd, true);
        if (k < 0)
                log_debug_errno(...); // propagate r below
}

return r;
```